tests/test_sync.py

```diff
@@ -257,8 +260,65 @@ def test_get_sync_work_state(self):
+
+        expected_results = {"good_blob_key": (good_blob_key, 7),  # good blobs have 7 parts, (e_tag takes 3)
+                            "bad_blob_key": (bad_blob_key, None),
+                            "missing_prefix": (missing_prefix, None),
+                            "gcp_parts": (gcp_parts, None)}
+
+        for k, v in expected_results.items():
+            match = BLOB_KEY_REGEX.match(v[0])
+            num_groups = len(match.groups()) if match else None
+            self.assertEquals(num_groups, v[1])
+
+    def test_skip_part_sync(self):
+        json_message = json.dumps({"bucket": "org-humancellatlas-dss-dev",
+                                   "componentCount": 30,
+                                   "contentType": "application/octet-stream",
+                                   "crc32c": "1n/pig==",
+                                   "etag": "CPjBmfqN1+QCEAE=",
+                                   "generation": "1568697601122552",
+                                   "kind": "storage#object",
+                                   "metageneration": "1",
+                                   "name": "blobs/"
+                                           "ea0b562ab28e22089283de9e691a5985d5118"
+                                           "fb567bab9e0ed11c1efd5d10995"
+                                           ".21a11df0cf55b1e197cf27d4185221dd9d909797."
+                                           "cbc62efcbe1a02c5b2f2fc38dce59287-933.5d0be7cf.partc",
+                                   "resourceState": "exists",
+                                   "selfLink": "https://www.googleapis.com/storage/v1/b/org-humancellatlas-dss-staging/"
+                                               "o/blobs"
+                                               "%2Ffd6f90b244d7bfe32978a5a7d51999d0eed60c5033465375e06ebbd921ba05b7."
+                                               "d0b75abd714f4a9792cdac50a074977bdc07f471."
+                                               "d69e4ed0099e33c1a546f9602fe82973-778.581f32a2.partc",
+                                   "size": "19658416372",
+                                   "storageClass": "MULTI_REGIONAL",
+                                   "timeCreated": "2019-09-17T05:20:01.122Z",
+                                   "timeStorageClassUpdated": "2019-09-17T05:20:01.122Z",
+                                   "updated": "2019-09-17T05:20:01.122Z"})
+        event = {"Records": [{
+                 "source_replica": "gcp",
+                 "dest_replica": "aws",
+                 "source_key": "blobs/ea0b562ab28e22089283de9e691a5985d5118fb567bab9e0ed11c1efd5d10995."
+                 "21a11df0cf55b1e197cf27d4185221dd9d909797.cbc62efcbe1a02c5b2f2fc38dce59287-933.5d0be7cf"
+                 ".partc",
+                 "body": json.dumps({"Message": json_message})
+                 }]
+                 }
+        results = daemon_app.launch_from_forwarded_event(event, None)
```

Comment:
shouldn't `daemon_app.launch_from_s3_event` be tested too?

Reply:
Although the .part blobs that we see are sourced from the gcp bucket, I've added the blob filter on the daemon_app.launch_from_s3_event as well for consistency.